pkg/github/github.go

```diff
@@ -418,46 +419,78 @@ func (g *GitHub) DownloadReleaseAssets(owner, repo string, releaseTags []string,
 		return errors.New("no release tags were populated")
 	}
 
-	for _, release := range releases {
-		releaseTag := release.GetTagName()
-		logrus.Infof("Download assets for %s/%s@%s", owner, repo, releaseTag)
+	funcs := []func() error{}
+	for i := range releases {
+		release := releases[i]
+		funcs = append(funcs, func() error {
+			releaseTag := release.GetTagName()
+			logrus.Infof("Download assets for %s/%s@%s", owner, repo, releaseTag)
+
+			assets := release.Assets
+			if len(assets) == 0 {
+				logrus.Infof("Skipping download for %s/%s@%s as no release assets were found", owner, repo, releaseTag)
+				return nil
+			}
 
-		assets := release.Assets
-		if len(assets) == 0 {
-			logrus.Infof("Skipping download for %s/%s@%s as no release assets were found", owner, repo, releaseTag)
-			continue
-		}
+			releaseDir := filepath.Join(outputDir, owner, repo, releaseTag)
+			if err := os.MkdirAll(releaseDir, os.FileMode(0o777)); err != nil {
+				return errors.Wrap(err, "creating output directory for release assets")
+			}
 
-		releaseDir := filepath.Join(outputDir, owner, repo, releaseTag)
-		if err := os.MkdirAll(releaseDir, os.FileMode(0o777)); err != nil {
-			return errors.Wrap(err, "creating output directory for release assets")
-		}
+			logrus.Infof("Writing assets to %s", releaseDir)
+			err := g.downloadAssetsParallel(assets, owner, repo, releaseDir)
+			if err != nil {
+				return err
+			}
 
-		logrus.Infof("Writing assets to %s", releaseDir)
+			return nil
+		})
+	}
 
-		for _, asset := range assets {
-			if asset.GetID() == 0 {
-				return errors.New("asset ID should never be zero")
-			}
+	return errorUtils.AggregateGoroutines(funcs...)
+}
 
+func (g *GitHub) downloadAssetsParallel(assets []github.ReleaseAsset, owner, repo, releaseDir string) error {
+	type assetFile struct {
+		File io.ReadCloser
+		Name string
+	}
+
+	done := make(chan assetFile, len(assets))
+	errch := make(chan error, len(assets))
+
+	for _, asset := range assets {
+		if asset.GetID() == 0 {
+			return errors.New("asset ID should never be zero")
+		}
+
+		go func(asset github.ReleaseAsset) {
 			logrus.Infof("GitHub asset ID: %v, download URL: %s", *asset.ID, *asset.BrowserDownloadURL)
```

Comment:
Can we use the go routine aggregator here as well?

Reply:
Done @saschagrunert 
thanks for your mentorship and review

Reply:
Thank you, I love it! :heart: :love_you_gesture: